subsys/bluetooth/shell/audio.c

```diff
@@ -1352,17 +1318,21 @@ static int cmd_init(const struct shell *sh, size_t argc, char *argv[])
 		return err;
 	}
 
-#if defined(CONFIG_BT_AUDIO_UNICAST)
+#if defined(CONFIG_BT_AUDIO_UNICAST_SERVER) || defined(CONFIG_BT_AUDIO_BROADCAST_SINK)
 	for (i = 0; i < ARRAY_SIZE(caps); i++) {
 		bt_audio_capability_register(&caps[i]);
 	}
+#endif /* CONFIG_BT_AUDIO_UNICAST || CONFIG_BT_AUDIO_BROADCAST_SOURCE */
 
-	/* Mark all supported contexts as available */
-	bt_audio_capability_set_available_contexts(BT_AUDIO_DIR_SINK,
-						   BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED);
-	bt_audio_capability_set_available_contexts(BT_AUDIO_DIR_SOURCE,
-						   BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED);
+	if (IS_ENABLED(CONFIG_BT_AUDIO_CAPABILITY)) {
+		/* Mark all supported contexts as available */
+		bt_audio_capability_set_available_contexts(BT_AUDIO_DIR_SINK,
+							   BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED);
```

Comment:
Maybe allow all types? If a client have "Media" content and you only allow "Unspecified" it will not be able to setup a stream.

Reply:
Agreed; updated